pkg/virt-operator/webhooks/kubevirt-update-admitter_test.go

```diff
@@ -64,4 +65,67 @@ var _ = Describe("Validating KubeVirtUpdate Admitter", func() {
 			},
 		}, 0),
 	)
+
+	table.DescribeTable("test validateWorkloadUpdateMethods", func(oldSpec, newSpec *v1.KubeVirtSpec, accept bool) {
+		causes := validateWorkloadUpdateMethods(oldSpec, newSpec)
+		if accept {
+			Expect(causes).To(BeEmpty())
+		} else {
+			Expect(causes).To(Not(BeEmpty()))
+		}
+	},
+		table.Entry("should accept LiveMigrate feature gate enabled and workloadUpdateMethod LiveMigrate",
+			&v1.KubeVirtSpec{},
+			&v1.KubeVirtSpec{
+				Configuration: v1.KubeVirtConfiguration{
+					DeveloperConfiguration: &v1.DeveloperConfiguration{
+						FeatureGates: []string{virtconfig.LiveMigrationGate},
+					},
+				},
+				WorkloadUpdateStrategy: v1.KubeVirtWorkloadUpdateStrategy{
+					WorkloadUpdateMethods: []v1.WorkloadUpdateMethod{v1.WorkloadUpdateMethodLiveMigrate},
+				},
+			},
```

Comment:
This could be overkill, but I think this would look better if you implemented a builder pattern:

```golang
type kubevirtSpecOption func(*v1.KubeVirtConfiguration)

func NewKubeVirtSpec(opts ...kubevirtSpecOption) *v1.KubeVirtConfiguration { ... }

func WithFeatureGate(featureGateName string) kubevirtSpecOption {
    ....
}

func WithWorkloadUpdateStrategy(strategy string) kubevirtSpecOption {
    ....
}
```
These would be reused below.

Reply:
It would look better indeed, I followed your suggestion, thanks!